### PR TITLE
[Perf] Optimize the Qwen2.5-Omni Model thinker-to-talker-proj with nn.Linear

### PR DESCRIPTION
**Pull request by kechengliu97**
# Summary of Changes

In this submission, we replaced the `thinker_to_talker_proj` from the original `ColumnParallelLinear` layer with `nn.Linear`. This change resulted in a significant performance improvement, particularly in the `forward` pass of the `Qwen2_5OmniTalkerForConditionalGeneration` model, where the overall execution time was reduced by over 200 microseconds.

# Performance Improvements

- **`thinker_to_talker_proj` latency**:
    
    - Reduced from `283μs 390ns` to `223μs 840ns`, a decrease of approximately 21%.
        
- **Overall `forward` time**:
    
    - The total execution time decreased by more than 200 microseconds, significantly improving inference performance.
        

# Configuration Comparison & Reference

In our comparison with the Transformer prototype, we observed that the corresponding linear layers also use `nn.Linear` in that model. Based on this observation, the updated implementation aligns with mainstream architectures, ensuring higher compatibility and optimizability.

# Precision Verification

Using the same prompt, "What is the origin of the United States?", we obtained identical outputs. Both the text and the wav file generated were almost identical, proving that the accuracy has not been affected by this change.

The output text (wav file read the following paragraph):
```
Prompt:
<|im_start|>system
You are Qwen, a virtual human developed by the Qwen Team, Alibaba Group, capable of perceiving auditory and visual inputs, as well as generating text and speech.<|im_end|>
<|im_start|>user
What is the origin of the United States?<|im_end|>
<|im_start|>assistant

vllm_text_output:
Well, you know, the United States has its roots in European colonization. The first Europeans to settle there were mainly from England. They started colonizing the eastern part of North America in the 17th century. People like the Pilgrims landed at Plymouth in 1620.Over time, these colonies grew and developed their own governments. Eventually, they declared independence from Britain in 1776. And then they formed the United States as we know it today.If you want to know more about specific events during that colonial period or how the states were formed, just let me know.
```

# Conclusion

By replacing `ColumnParallelLinear` with `nn.Linear`, we not only improved performance but also ensured consistency with Transformer architecture. This optimization will result in lower latency and higher throughput for various inference tasks, with no loss in precision.

### CODE REVIEW

File: vllm_omni/model_executor/models/qwen2_5_omni/qwen2_5_omni_talker.py [hunk 1]
```diff
@@ -68,13 +68,9 @@ def __init__(self, *, vllm_config: VllmConfig, prefix: str = ""):
...
 
-        self.thinker_to_talker_proj = ColumnParallelLinear(
+        self.thinker_to_talker_proj = nn.Linear(
             self.config.embedding_size,
             self.config.hidden_size,
-            bias=True,
-            gather_output=True,
-            skip_bias_add=False,
-            quant_config=quant_config,
         )
```
[the remark this answers is not in the document]
Comment (kechengliu97) [reply vllm_omni/model_executor/models/qwen2_5_omni/qwen2_5_omni_talker.py:73]: Meanwhile, Qwen-2.5-Omni-7B is a small model which almost all kinds of NPU/GPU can hold with only one.

**Comment by hsliuustc0106** [comment]
any accuracy difference?

**Comment by kechengliu97** [comment]
> any accuracy difference?

With the pure text prompt as input, no difference has been observed so far. More importantly, this is what is used in the corresponding part of the **Transformer** .

**Comment by ZJY0516** [comment]
@kechengliu97 Do you have any idea that where the performance improvements come from? `ColumnParallelLinear ` call nn.linear inside actually

**Comment by kechengliu97** [comment]
> @kechengliu97 Do you have any idea that where the performance improvements come from? `ColumnParallelLinear ` call nn.linear inside actually

Our Device can not transfer the profiling swim lane to the outer Internet, but I noticed that using these like ColumnParallelLinear or RowParallelLinear we found some communication cost, which may cause the higher latency.
